strapi/strapi
# [Custom fields] Add custom fields registry in admin core

### What does it do?

- Adds a registry to store custom fields in Strapi Admin
- Adds a method to add an item to that registry
- Expose the stored custom fields in react context
- Update the example plugin to test the registry

### How to test it

- yarn build on the monorepo root (necessary because of changes to the helper plugin)
- Run yarn develop --watch-admin on getstarted and open the app
- Modify any admin page to log the reponse of useCustomFields (hook coming from the helper plugin):

```js
import { useCustomFields } from '@strapi/helper-plugin';

const AnyComponent = () => {
  // ...
  const { customFields } = useCustomFields();
  console.log(customFields);
  // ...
}
```

- Check the log in the console. It should be an object with `plugin::mycustomfields.color` as the key, and the object passed to `customFields.register()` as the value

Additional checks when modifying the register call in mycustomfields:

- Modify the register() call in mycustomfields to remove the pluginId. The key in the logged object should now be `global::global.color`
- Remove either the name, type, intlLabel, intlDescription or components, and you should get an error preventing the admin from starting

### Related links

[Custom fields RFC](https://github.com/strapi/rfcs/blob/custom-fields-api/rfcs/xxxx-custom-fields-api.md#admin)
